app/Providers/FortifyServiceProvider.php

```diff
@@ -149,42 +152,56 @@ public function boot(): void
 
             $user = User::query()->where('username', $request->username)->first();
 
+            if ($user && !Hash::check($request->password, $user->password)) {
```

Comment:
We shouldn't be hashing the password twice. If we did, we should consider increasing the default bcrypt value. This should be memoized with the following if statement.